registration/src/main/java/org/spine3/samples/lobby/registration/order/OrderAggregate.java

```diff
@@ -0,0 +1,391 @@
+                    .addAllSeat(command.getSeatList());
+            return result.build();
+        }
+
+        private static OrderReservationCompleted buildOrderReservationCompleted(MarkSeatsAsReserved command) {
+            final OrderReservationCompleted.Builder result = OrderReservationCompleted.newBuilder()
+                    .setOrderId(command.getOrderId())
+                    .setReservationExpiration(command.getReservationExpiration())
+                    .addAllSeat(command.getSeatList());
+            return result.build();
+        }
+    }
+
+    private static class Validator {
+
+        private static void checkNotConfirmed(Order order, Message cmd) {
+            final String message = format("Cannot modify a confirmed order with ID: %s; command: %s.",
+                    order.getId().getUuid(), cmd.getClass().getName()
+            );
+            checkState(!order.getIsConfirmed(), message);
+        }
+
+        private static void checkNewState(Order order) {
+            checkState(order.hasId(), "No order ID in a new order state.");
+            final String orderId = order.getId().getUuid();
+            checkState(order.hasConferenceId(), "No conference ID in a new order state, ID: " + orderId);
+            if (order.getSeatList().isEmpty()) {
+                throw new IllegalArgumentException("No seats in a new order state, ID: " + orderId);
+            }
+        }
+
+        private static void validateCommand(RegisterToConference cmd) {
+            checkOrderId(cmd.hasOrderId(), cmd);
+            checkField(cmd.hasConferenceId(), "conference ID", cmd);
+            checkSeats(cmd.getSeatList(), cmd);
+        }
+
+        private static void validateCommand(MarkSeatsAsReserved cmd) {
+            checkOrderId(cmd.hasOrderId(), cmd);
+            checkField(cmd.hasReservationExpiration(), "reservation expiration", cmd);
+            checkSeats(cmd.getSeatList(), cmd);
+        }
+
+        private static void validateCommand(RejectOrder cmd) {
+            checkOrderId(cmd.hasOrderId(), cmd);
+        }
+
+        private static void validateCommand(ConfirmOrder cmd) {
+            checkOrderId(cmd.hasOrderId(), cmd);
+        }
+
+        private static void validateCommand(AssignRegistrantDetails cmd) {
+            checkOrderId(cmd.hasOrderId(), cmd);
+            checkField(cmd.hasRegistrant(), "registrant", cmd);
+        }
+
+        private static void checkOrderId(boolean hasId, Message cmd) {
+            checkField(hasId, "order ID", cmd);
+        }
+
+        private static void checkSeats(List<SeatQuantity> seats, Message cmd) {
+            checkField(!seats.isEmpty(), "seats", cmd);
+        }
+
+        private static void checkField(boolean hasField, String fieldName, Message cmd) {
+            if (!hasField) {
+                throw new IllegalArgumentException("No " + fieldName + " in the command: " + cmd.getClass().getName());
```

Comment:
How about this wording: "The field 'fieldName' must be defined in all commands of class <class>."?